packages/core/utils/src/dal/mikro-orm/base-entity.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  BeforeCreate,
+  Entity,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+} from "@mikro-orm/core"
+import { generateEntityId } from "../../common"
+
+@Entity({ abstract: true })
+export class BaseEntity {
+  [OptionalProps]?: BaseEntity["id"] | BaseEntity["__prefix_id__"]
+
+  private __prefix_id__?: string
+
+  constructor({ prefix_id }: { prefix_id?: string } = {}) {
+    this.__prefix_id__ = prefix_id
+  }
+
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @OnInit()
+  @BeforeCreate()
+  onInit() {
+    this.id ??= this.generateEntityId(this.__prefix_id__)
+  }
+
+  private generateEntityId(prefixId?: string): string {
+    let ensuredPrefixId: string = prefixId as string
+
+    if (ensuredPrefixId) {
+      return generateEntityId(undefined, ensuredPrefixId)
+    }
+
+    ensuredPrefixId = Object.getPrototypeOf(this).constructor.name
+
+    /*
+     * Split the class name (camel case) into words and exclude model and entity from the words
+     */
+    const words = ensuredPrefixId
+      .split(/(?=[A-Z])/)
```

Comment:
Might be insignificant, but it would be nice to cache this calculation so we don't have to run regex every time an entry is created.

Reply:
done 👍